### PR TITLE
[WIP] Documentation

### PR DESCRIPTION
**Pull request by vitvakatu**
- [x] Provide top-level documentation for every struct.
- [x] Provide documentation for every entity, field and method.
- [ ] Provide in-documentation examples.

### CODE REVIEW

File: src/scene.rs [hunk 1]
```diff
@@ -204,6 +237,9 @@ impl AmbientLight {
     }
 }
 
+/// The light source that illuminates all objects equally from a given direction,
+/// like an area light of infinite size and infinite distance from the scene;
```
Comment (kvark) [inline src/scene.rs]: I don't think "infinite size" needs to be here

**Comment by kvark** [comment]
The in-documentation examples is something I'm not 100% sure about, since even if you make them CI-tested, they still add up the maintenance cost, especially now that the library is still evolving rapidly.

File: src/scene.rs (hunk 1 above)
Comment (vitvakatu) [inline src/scene.rs]: So, only "The light source that illuminates all objects equally from a given direction"?

Comment (kvark) [inline src/scene.rs]: that would be fine, yeah

File: src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,22 @@
+#![warn(missing_docs)]
+//! # Motivation and Goals
+//!
+//! 1. Ergonomics is first priority. Being able to prototype quickly and code
```
Comment (kvark) [inline src/lib.rs]: I'm not keen on the idea of duplicating this both here and in the Readme...

Comment (vitvakatu) [inline src/lib.rs]: Oh, yeah, I forgot to say that. I need help to write good module-level documentation :D

Comment (kvark) [inline src/lib.rs]: I think not having any is somewhat better than duplication, so it's fine to remove that.

Comment (vitvakatu) [inline src/lib.rs]: But we have `#![warn(missing_docs)]`. Should I remove that? (I've documented everything)

Comment (kvark) [inline src/lib.rs]: No, just have a one-liner for the module documentation

**Comment by kvark** [comment]
See, now it's trying to test your in-doc tests. At this point, we need to either implement OsMesa support for headless rendering, or not have those in-doc examples at all.

**Pull request merged**